lib/gman_client.rb

```diff
@@ -69,9 +78,16 @@ def retrieve_token
       JSON.parse(response)['access_token']
     end
 
+    private
+
     def convert_payload(response)
-      response.payload.first.nil? ? [response.to_h] : response.map(&:to_h)
+      response.map do |hash_request|
+        Hashie.symbolize_keys!(hash_request.to_h)
+      end
+    end
+
+    def attempt_thrice
```

Comment:
How about renaming to `attempt(retry_count)` and passing `retry_count` to `:max_tries`?